ui/app/src/components/DatasourceList/DatasourceDrawer.tsx

```diff
@@ -12,53 +12,36 @@
 // limitations under the License.
 
 import { Datasource } from '@perses-dev/core';
-import { DispatchWithoutAction, useState } from 'react';
-import { DiscardChangesConfirmationDialog, Drawer, ErrorAlert, ErrorBoundary } from '@perses-dev/components';
+import { DispatchWithoutAction, useCallback, useState } from 'react';
+import { Drawer, ErrorAlert, ErrorBoundary } from '@perses-dev/components';
 import { PluginRegistry } from '@perses-dev/plugin-system';
 import { bundledPluginLoader } from '../../model/bundled-plugins';
 import { DeleteDatasourceDialog } from '../dialogs/DeleteDatasourceDialog';
 import { DatasourceEditorForm } from './DatasourceEditorForm';
 
+export type Action = 'Create' | 'Update';
+
 interface DatasourceDrawerProps {
   datasource: Datasource;
   isOpen: boolean;
-  saveAction: string;
+  saveAction: Action;
   onSave: (datasource: Datasource) => void;
   onClose: DispatchWithoutAction;
 }
 
 export function DatasourceDrawer(props: DatasourceDrawerProps) {
   const { datasource, isOpen, saveAction, onSave, onClose } = props;
   const [isDeleteDatasourceDialogStateOpened, setDeleteDatasourceDialogStateOpened] = useState<boolean>(false);
-  const [isDiscardDialogStateOpened, setDiscardDialogStateOpened] = useState<boolean>(false);
-  const [hasDraftChanges, setHasDraftChanges] = useState<boolean>(false);
-
-  const handleSave = (datasource: Datasource) => {
-    setHasDraftChanges(false);
-    onSave(datasource);
-  };
-
-  const handleClose = () => {
-    if (hasDraftChanges) {
-      setDiscardDialogStateOpened(true);
-    } else {
-      onClose();
-    }
-  };
 
-  const handleDelete = () => {
-    setHasDraftChanges(false);
-    onClose();
-  };
-
-  const handleDiscard = () => {
-    setHasDraftChanges(false);
-    setDiscardDialogStateOpened(false);
-    onClose();
-  };
+  // When user clicks out of the drawer, do not close it, just do nothing
+  // This is a quick-win solution to avoid losing draft changes
+  // -> TODO allow closing by clicking-out without being too sensitive to missclicks
+  const handleClickOut = useCallback(() => {
```

Comment:
As discussed in another place with @celian-garcia, moving the discard change logic into the DatasourceEditorForm component made it more complex to tackle the "click out of drawer" case, on which we would like the same behavior as the cancel button. In the current state it would mean moving around a datasource object between the drawer & the form component to achieve json comparison at the drawer level, which we would like to avoid.. Letting @celian-garcia think about a better solution in a next PR, for now going with this quick-win of just disabling it.

Reply:
ack :+1: I will propose a refactor of that in the following days